src/Adapter/MSTest.CoreAdapter/Discovery/TypeEnumerator.cs

```diff
@@ -204,6 +204,15 @@ internal UnitTestElement GetTestFromMethod(MethodInfo method, bool isDeclaredInT
                 this.type,
                 warnings);
 
+            // get DisplayName from TestMethodAttribute
+            var myAttribute = method.GetCustomAttributes(false)
+                                    .OfType<TestMethodAttribute>()
```

Comment:
Should we be using reflecthelper to get custom attributes here like other instances in this method? That should help with unit testing as well. And reflecthelper abstracts this logic out for different platforms.

Reply:
Good call. Fixed.